File: test/nodejs/oml_test.js
var duckdb = require('../../duckdb/tools/nodejs');
var assert = require('assert');

describe(`oml extension`, () => {
    let db;
    let conn;
    before((done) => {
        db = new duckdb.Database(':memory:', {"allow_unsigned_extensions":"true"});
        conn = new duckdb.Connection(db);
        conn.exec(`LOAD '${process.env.OML_EXTENSION_BINARY_PATH}';`, function (err) {
            if (err) throw err;
            done();
        });
    });

    it('oml function should return expected amount of rows', function (done) {
        db.all("SELECT * from OmlGen('data/oml_testing/st_lrwan1_11.oml');", function (err, res) {
            if (err) throw err;
            assert.deepEqual(res, [{value: 67725}]);
            done();
        });
    });
});

File: test/python/oml_test.py
import duckdb
import os
import pytest

# Get a fresh connection to DuckDB with the oml extension binary loaded
@pytest.fixture
def duckdb_conn():
    extension_binary = os.getenv('OML_EXTENSION_BINARY_PATH')
    if (extension_binary == ''):
        raise Exception('Please make sure the `OML_EXTENSION_BINARY_PATHH` is set to run the python tests')
    conn = duckdb.connect('', config={'allow_unsigned_extensions': 'true'})
    conn.execute(f"load '{extension_binary}'")
    return conn

def test_oml(duckdb_conn):
    duckdb_conn.execute("SELECT * from OmlGen('data/oml_testing/st_lrwan1_11.oml');");
    res = duckdb_conn.fetchall()
    assert(res[0][0] == 67725);